library/ssl_tls.c

```diff
@@ -7309,10 +7309,13 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl,
     mbedtls_x509_crt *ca_chain;
     mbedtls_x509_crl *ca_crl;
 
-    ssl->handshake->peer_authenticated = MBEDTLS_SSL_FI_FLAG_UNSET;
     if( authmode == MBEDTLS_SSL_VERIFY_NONE )
+    {
+        ssl->handshake->peer_authenticated = MBEDTLS_SSL_FI_FLAG_SET;
```

Comment:
Not very enthusiastic about that either. Same as for session resumption: I'd rather not set this flag when authentication has not been done, but instead just not check it when we don't need to check it.